enumerable.rb

```diff
@@ -0,0 +1,149 @@
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < results.length
+        yield(results[i])
+        i += 1
+      end
+      results
+    else
+      results.to_enum
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < results.length
+        yield(results[i], i)
+        i += 1
+      end
+      results
+    else
+      results.to_enum
+    end
+  end
+
+  def my_select
+    if block_given?
+      choice = []
+      results.my_each do |x|
+        if yield(x)
+          choice.push(x)
+        end
+      end
+      choice
+    else
+      results.to_enum
+    end
+  end
+
+  def my_all?(items = nil)
+    if block_given?
+      my_each { |x| return false unless yield x }
+    else
+      return my_all? { |obj| obj } unless items
+
+      if items.class == Regexp
+        my_each { |x| return false unless items.match?(x) }
+      elsif items.class == Class
+        my_each { |x| return false unless x.is_a? items }
+      else
+        my_each { |x| return false unless x == items }
+      end
+    end
+    true
+  end
+
+  def my_any?(items = nil)
+    result = false
+    if block_given?
+      my_each { |item| result = true if yield item }
+    elsif items
+      my_each { |item| result = true if itemstern?(item, items) }
+    else
+      my_each { |item| result = true if item }
+    end
+    result
+  end
+
+  def my_none?(items = nil)
+    result = true
+    if block_given?
+      my_each { |item| result = false if yield item }
+    elsif items
+      my_each { |item| result = false if itemstern?(item, items) }
+    else
+      my_each { |item| result = false if item }
+    end
+    result
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    if block_given?
+      my_each { |x| count += 1 if yield(x) }
+    elsif arg
+      my_each { |x| count += 1 if x == arg }
+    else
+      count = length
+    end
+    count
+  end
```

Comment:
- [ ] counts the number of items in enum that are equal to item if an argument is given
- [ ] counts the number of elements yielding a true value if a block is given